asv_bench/benchmarks/rolling.py

```diff
@@ -150,19 +150,23 @@ def time_quantile(self, constructor, window, dtype, percentile, interpolation):
         self.roll.quantile(percentile, interpolation=interpolation)
 
 
-class PeakMemFixed:
-    def setup(self):
-        N = 10
+class PeakMemFixedWindowMinMax:
+
+    params = [
+        pd.core.window.rolling.Rolling.min,
+        pd.core.window.rolling.Rolling.max,
+    ]
+
+    def setup(self, f):
+        N = int(1e5)
         arr = 100 * np.random.random(N)
-        self.roll = pd.Series(arr).rolling(10)
-
-    def peakmem_fixed(self):
-        # GH 25926
-        # This is to detect memory leaks in rolling operations.
-        # To save time this is only ran on one method.
-        # 6000 iterations is enough for most types of leaks to be detected
-        for x in range(6000):
-            self.roll.max()
+        self.roll = pd.Series(arr).rolling(1000)
+
+    def peakmem_fixed(self, f):
+        # GH 33693
+        # increased size of array so we see a clearer picture
+        for x in range(1000):
```

Comment:
make some part of this smaller
22s is way too long 

we have thousands of benchmarks in the suite

your goal time should <1s